cmd/catalyst-node/catalyst-node.go

```diff
@@ -865,39 +868,58 @@ func (ac *PlaybackAccessControl) handleUserNew(payload []byte) []byte {
 	return []byte("false")
 }
 
-func (ac *PlaybackAccessControl) getPlaybackAccessControlInfo(playbackID, pubKey string) (map[string]*PlaybackAccessControlEntry, error) {
+func getPlaybackAccessControlInfo(ac *PlaybackAccessControl, playbackID, pubKey string) (map[string]*PlaybackAccessControlEntry, error) {
 	if ac.cache[playbackID] == nil ||
 		ac.cache[playbackID][pubKey] == nil ||
 		time.Now().After(ac.cache[playbackID][pubKey].Stale) {
 
-		err := ac.cachePlaybackAccessControlInfo(playbackID, pubKey)
+		err := cachePlaybackAccessControlInfo(ac, playbackID, pubKey)
 		if err != nil {
 			return nil, err
 		}
 
 	} else if time.Now().After(ac.cache[playbackID][pubKey].MaxAge) {
-		go ac.cachePlaybackAccessControlInfo(playbackID, pubKey)
+		go cachePlaybackAccessControlInfo(ac, playbackID, pubKey)
 	}
 
 	return ac.cache[playbackID], nil
 }
 
-func (ac *PlaybackAccessControl) cachePlaybackAccessControlInfo(playbackID, pubKey string) error {
+func cachePlaybackAccessControlInfo(ac *PlaybackAccessControl, playbackID, pubKey string) error {
 	body, err := json.Marshal(PlaybackAccessControlRequest{"jwt", pubKey, playbackID})
 	if err != nil {
 		return err
 	}
 
-	req, err := http.NewRequest("POST", ac.gateURL, bytes.NewReader(body))
+	statusCode, maxAge, stale, err := queryGate(ac, body)
 	if err != nil {
 		return err
 	}
 
+	var allow = statusCode == 204
```

Comment:
**nit:** I guess this boolean could be the return type of `queryGate`. We don't use the `statusCode` here elsewhere.

Reply:
less nit: We should actually allow other `2xx` responses as success here. Would change this to either
```suggestion
	var allow = statusCode/100 == 2
```
or
```suggestion
	var allow = statusCode >= 200 && statusCode < 300
```